### ISSUE TITLE
Hike difficulty calculator

### ISSUE THREAD

**Issue by cdusqud** [Maproom/qmapshack#281]
**Important Notice:**

**Do not delete any of the sections. Read what is requested.**
**_Replace(!)_ the instructions by your _answer(!)_. Feature requests**
**reports that are not properly filed might be closed without**
**further notice.**

**Choose a good title**
[X ] Did you choose a good title?

**Is your feature request related to a problem? Please describe.**

One of the most important factors to consider when you're creating a new hiking track is trying to figure out how difficult your track will actually be. This can be, of course, be calculated mentally, based on previous experience, but there are several methods over there to calculate the estimated difficulty of a track automatically. Some of them are complex, take in consideration many different factors and uses unknown formulas, but some of them are really simple and use very simple formulas and scales. 

Maybe two of the most well known and simplest methods of this kind are the "Shenandoah National Park" method and the Paul  Petzoldt's method. Both take in consideration only two variables  (distance and elevation gain) and both do their job pretty well, in my opinion. 

The actual formulas are:
-Shenandoah National Park index: (Square Root of ((Elevation Gain in feet X 2) X Distance in miles))
https://www.nps.gov/shen/planyourvisit/how-to-determine-hiking-difficulty.htm#:~:text=Shenandoah's%20Hiking%20Difficulty%20is%20determined,root%20is%20the%20numerical%20rating.

-Petzoldt's index: (Distance in miles + (Elevation Gain in feet / 500))
http://www.tetonhikingtrails.com/about-us.htm

**Describe the solution you'd like**
It would be great if QMS would offer an estimation of the track difficulty. This estimation, could be based on Shenandoah method, on Petzoldt method, or on a combination of both methods + more factor, see for example:

https://trailsnh.com/tools/hiking-difficulty.php

Of course, other -better and more complex- methods could be also used, I'm just giving two basic examples.

Ideally (and optionally), difficulty index could be shown on the track metric window, so the user can have an idea of the estimated difficulty of the track while still on track edit mode:
![1  Diff  est  in Track Edit Mode](https://user-images.githubusercontent.com/76106623/102487272-91b1ed00-406a-11eb-82d3-e5868a3636e7.png)

Additionally, it could be also implemented as a new filter to be used with already created tracks:
![2  New Filter](https://user-images.githubusercontent.com/76106623/102487321-a8584400-406a-11eb-8302-2cf04704d36b.png)

But it also could be integrated as and additional option into the already existing "Energy Use Cycling" or "Change Speed" filters. Maybe this would be the best option, but of course, the name of the filters should be properly changed in this case. 

**Describe alternatives you've considered**
An alternative is using an user created spreadsheet, or one of the several web sites that offer this kind of information.

**Give a reason why someone else should spend time on that request**
An estimation of the hiking difficulty of a track would be an useful feature and a logical addition to the already existent "Change Speed" and "Energy use cycling" filters.

**Additional context**
There are excellent web sites offering this kind of information, but some of them requires importing a gpx file (IBP index), and some of them requires unit conversion (miles to km, feet to meter):
https://www.ibpindex.com/index.php/en
https://trailsnh.com/tools/hiking-difficulty.php
http://www.nwhiker.com/HikeEval.html
The basic idea would be having a basic estimation of the difficult of the track being created on the fly, directly within the program without the need to go to third party webs.

**Comment by kiozen** [comment]
Nice idea! Here my two cents for anyone implementing this:

This kind of information is always very specific to a small sub-group of users. Same as with the energy consumption calculations for cyclists. And therefore it's always kind of hard to decide how to configure it. Naturally configuration becomes a bit hellish.

So what's missing is a good UX concept on how to configure the stuff and how to pull in the existing  energy consumption calculations for cyclists. So it's not that trivial.

Personally I know my benchmarks on ascent/descent and distances therefore  I never missed a score.

**Comment by mitxel-m** [comment]
I am very sceptical about these formulas and what they can bring.

First of all, one should not confuse difficulty with effort. These formulas give a value of effort, and then translate it into difficulty, which IMHO is a serious error.


I thnk that these formulas can be used for comparing various paths of a similar nature, such as those marked and maintained in a certain area, as highlighted on the National Park website, and are only useful to inform tourist or ocasinnally hikers, but I find them inapplicable for most of the mountain paths you will see on a map.

A clear and simple example in short words: Assume you have two walking routes from start A to top B, so same elevation gain but different distance. As long as the formulas take in account the total elevation gain and the total distance,  the "direttissima route" results easier and requires less effort than the "normal route".  Really?

I have tried our with several known tracks, and have obtained inconsistent results, which can lead to confusion and wrong decisions.
 
 They don't not take into account what is in the middle of the route. So if you compare two options with this formule,  the option indicated as requiring less effort is not always the easiest or the one that requires less " power ", because they are ignoring  slope changes along the route.

To mitigate this, some of the mentioned pages add a steepness gradient data to the Shenadon formule  but  it fails again as it only takes the total elevation gain and distance between start and end point. Thus the option indicated as requiring less effort is not always the easiest or the one that requires less " power ", because they are ignoring  slope changes along the route.

In addition, they do not take into account that the difficulty on the descent increases if the path is very steep, and sometimes requires more effort than a flat path.


Some real exmples:

-1 Mount Higa. 

![](https://i.ibb.co/jDCwBTP/Selecci-n-016.jpg)

The formula says the pink one the easiest and less effort. However, it is by far the most difficult. It is a path where a section  is very steep and requires skill to climb high steps. On the contrary, it marks the green one as the most difficult when in fact it is easier. It is a track to the tower. Result-> wrong decission
Coloring the slope gives you better information.
![](https://i.ibb.co/r7s0pxG/Selecci-n-015.jpg)


-2 Anboto

The formula says the red one the easiest and less efort. In fact it is more exposed, and requires a solid experience. Someone could get in serious trouble. The reasom is that difficulty is not only elevation gain and distance. 
![](https://i.ibb.co/xzc6syC/Selecci-n-014.jpg)

and the same in the other side of the same mountain. The blue one is the normal route. but the formula syas that the pink one is the easiest...

![](https://i.ibb.co/drbvJtq/Selecci-n-011.jpg)

---

**Conclusion** (personal): I find that is not a good idea to use these formulas  to plan a route. as long that they are inconsistent and do not fit in any way to reality, 

**Alternatives:** 
If you want to get more realistic data when planning your own routes, it should be more useful an option in the route drawing that colours the slope of the track on the fly, instead of a simple colour. This way you can see at a glance the character of the section you are going to incorporate into the route. This what is proposed at #306

Another way would be play with the router and brouter profiles to use your own criteria (or rely on the pre-set ones).

**Comment by pingurus** [comment]
> Another way would be play with the router and brouter profiles to use your own criteria (or rely on the pre-set ones).

If you create a route (not a track!) with brouter, you even see what the cost that brouter attributed to this route is. By that you could compare difficulty between two routes, provided the metric used in brouter corresponds to your personal 'feeling' for difficulty.

**Comment by kiozen** [comment]
Personally I wouldn't have much use of such a feature. I know my personal limits (ascent, descent, distance) pretty well. No formula needed for that one. And difficulty by terrain is always object to research and personal capabilities.

**Comment by cdusqud** [comment]
Yes, I agree that those users who know well their personal limits would hardly find any use in a feature of this kind. I also fully agree with the fact that there are no "magic" formulas, and that the examples I gave (Shenandoah and Petzoldt indexes), while working reasonably well in (let's say) "Standard" hiking tracks, might not work so well (or at all) in some scenarios, for example, when applied to climbing routes with short horizontal distance & high elevation gain. 
However, I gave these two particular examples just because they are well known, really simple, and work well in many "standard" situations. Of course better, more complex algorithms are always  possible. The main idea, in other words, would be a kind of IBP index built in QMS, which I still think could be an useful feature for many users as a reference for quick comparison between different tracks. 
https://www.ibpindex.com/index.php/en/

Anyway, thanks everyone for commenting here and for your great, great work with QMS.
